R/verb-select.R

```diff
@@ -83,89 +83,76 @@ rename_with.tbl_lazy <- function(.data, .fn, .cols = everything(), ...) {
 #' @inheritParams dplyr::relocate
 #' @export
 relocate.tbl_lazy <- function(.data, ..., .before = NULL, .after = NULL) {
-  vars <- simulate_vars(.data)
   new_vars <- dplyr::relocate(
-    simulate_vars(.data),
+    simulate_vars2(.data),
     ...,
     .before = {{.before}},
     .after = {{.after}}
   )
-  .data$ops <- op_select(.data$ops, syms(set_names(names(new_vars))))
+
+  .data$lazy_query <- add_select(.data, syms(unlist(new_vars)))
   .data
 }
 
 simulate_vars <- function(x) {
   as_tibble(rep_named(op_vars(x), list(logical())))
 }
 
+simulate_vars2 <- function(x) {
```

Comment:
This is needed because `relocate()` can also rename variables. This didn't work before

Reply:
I'm not sure I understand the difference here. Doesn't this produce a 1-row tibble with the names of the variables in the cell values? Why doesn't it need `.name_repair` like `simulate_vars()`?